src/main/engine/Renderer.h

```diff
@@ -3,7 +3,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-#include "Engine.h"
+class Engine;
```

Comment:
We don't actually need to know about the contents of the Engine class from inside this header.